src/pip/_internal/index/package_finder.py

```diff
@@ -219,6 +227,30 @@ def evaluate_link(self, link: Link) -> tuple[LinkType, str]:
 
                 version = wheel.version
 
+        # Check upload-time filter after verifying the link is a package file.
+        # Skip this check for local files, as --uploaded-prior-to only applies
+        # to packages from indexes.
+        if self._uploaded_prior_to is not None and not link.is_file:
+            if link.upload_time is None:
+                if isinstance(link.comes_from, IndexContent):
+                    index_info = f"Index {link.comes_from.url}"
+                elif link.comes_from:
+                    index_info = f"Index {link.comes_from}"
+                else:
+                    index_info = "Index"
```

Comment:
I'm afraid that I'm not familiar with the link class. When would the `comes_from` attribute be a string vs a IndexContent object? I'm just worried that this may not return the index URL in certain edge cases.